common/src/main/java/com/microsoft/identity/common/internal/cache/MsalCppOAuth2TokenCache.java

```diff
@@ -147,8 +205,20 @@ public synchronized void clearCache() {
     public synchronized AccountDeletionRecord removeAccount(@NonNull final String homeAccountId,
                                                             @Nullable final String environment,
                                                             @NonNull final String realm) throws ClientException {
+        // TODO This API is potentially problematic for TFW/TFL...
+        // Normally on Android, apps are 'sandboxed' such that each app has their own cache
+        // and we don't have to worry about 1 app stomping on another's cache
+        //
+        // TFW/TFL however, "double stacked" their app registrations into a single binary
+        // Such that calling removeAccount() will potentially remove the Account being used by
+        // another app.
+        //
+        // This API assumes the *general* case where an app is single stacked. If special
+        // accommodations need to come later for Teams then we can reevaluate the logic here.
+
```

Comment:
This cache `MsalCppOauth2TokenCache` is local cache, right? Like this is when local auth is happening and not broker?

Reply:
Correct, this is always local cache -- not a broker

Reply:
Resolving this; feel free to reactivate if you have any more questions

Reply:
In that case, this shouldn't be problematic, right? TFW and TFL are two different packages right?

Reply:
Nope, they are a single binary. 2 app registrations shipped inside of a single app from the app store. 

If a user switches between them, they use the same cache on disk because they are for Android's purposes "the same app"

Reply:
It's configured somewhat similarly to our testapp; Teams chooses which app.json to load and which client_id to use based on if the user is enterprise vs. consumer

Reply:
Got it, Thanks!